metl/model_encoder.py

```diff
@@ -0,0 +1,114 @@
+import torch
+from typing import Literal
+from biopandas.pdb import PandasPdb
+import metl.relative_attention as ra
+from Bio.SeqUtils import seq1
+import os
+
+class ModelEncoder(torch.nn.Module):
+    def __init__(self, model, encoder, strict=True, indexing:Literal[0,1] = 0) -> None:
+        super(ModelEncoder, self).__init__()
+
+        if indexing != 0 and indexing != 1:
+            raise Exception("Indexing must be equal to 0 or to 1.")
+
+        self.model = model
+        self.encoder = encoder
+
+        self.indexing = indexing
+        self.strict = strict
+
+        self.needs_pdb = self.check_if_pdb_needed(model)
+
+    def check_if_pdb_needed(self, model):
+        sequential = next(model.children())
+        
+        for layer in sequential:
+            if isinstance(layer, ra.RelativeTransformerEncoder):
+                return True
+        return False
+
+    def validate_pdb(self, pdb_file, wt):
+        try:
+            ppdb = PandasPdb().read_pdb(pdb_file)
+        except Exception as e:
+            raise Exception(f"{str(e)} \n\n PDB file could not be read by PandasPDB. It may be incorrectly formatted.")
+
+        groups = ppdb.df['ATOM'].groupby('residue_number')
```

Comment:
Sam, have you worked with multi-chain PDB files. Do you anticipate any complications there?

Reply:
@samgelman

Reply:
I have, and a multi-chain PDB file could potentially cause a problem here. I need to investigate how BioPandas handles multiple chains.

Reply:
The most user-friendly fix would to have both documentation stating single chain structures are expected and validation code confirming a single chain structure is provided.

Reply:
@agitter Is METL compatible with multi-chain PDB files? If not, it would be best to clarify that when reading the file. I don't have one of those files though so I can't really see what in the library would be used to check if the PDB file is single or multi-chain.

Reply:
METL is not compatible with multi-chain PDB files. We want single chains. 

The metl-sim repo has a function that reads a sequence from a PDB file and gives an error on multiple chains:
https://github.com/gitter-lab/metl-sim/blob/f94ed68e2ba6c3c009ab0825c8b035a1ac82c46f/code/utils.py#L91

Reply:
@John-Peters-UW let's grab that code that is used to `raise ValueError("PDB contains more than one chain: {}".format(pdb_fn))` in the file Sam linked. Then we'll need to think about anywhere in the metl-pretrained docs to state that requirement for single chain PDB files.

Reply:
That would work. I can look at that the week of the 23rd. I'm gone next week for a conference in Boston but I'll have time afterwards